File: Python/Level 2/수식 최대화 (구현).py
import re,copy

def solution(expression):
    num = re.split(r'\W+',expression)           # /W 는 알파벳,숫자,'_' 가 아닌 문자를 의미함
    car = re.split(r'\w+',expression)[1:-1]     # /w 는 알파벳,숫자,'_' 인 문자를 의미함
    order = [
        ['+','-','*'],
        ['+','*','-'],
        ['-','+','*'],
        ['-','*','+'],
        ['*','+','-'],
        ['*','-','+']
    ]
    ans = 0
    
    for i in order:
        copynum = list(map(int,copy.deepcopy(num)))
        copycar = copy.deepcopy(car)
        
        for j in i:
            while j in copycar:
                index = copycar.index(j)
                
                if j == '+':
                    value = copynum[index] + copynum[index+1]
                elif j == '-':
                    value = copynum[index] - copynum[index+1]
                else:
                    value = copynum[index] * copynum[index+1]
                    
                copycar.remove(j)
                copynum[index] = value
                del copynum[index+1]
        
        ans = max(ans,abs(copynum[0]))
    
    return ans

File: Javascript/Level 2/수식 최대화 (구현).js
function solution(expression) {
	let answer = 0;
	let num = expression.split(/[^0-9]/g).map((v) => Number(v)); // 정규식으로 숫자만 분리
	let sign = [];
	let orders = [
		["-", "+", "*"],
		["-", "*", "+"],
		["+", "-", "*"],
		["+", "*", "-"],
		["*", "+", "-"],
		["*", "-", "+"],
	];

	// 부호만 분리해서 sign에 저장
	expression.split("").forEach((v) => {
		if (orders[0].indexOf(v) !== -1) {
			sign.push(v);
		}
	});

	orders.forEach((order) => {
		let copy_num = num.slice();
		let copy_sign = sign.slice();

		for (let i = 0; i < 3; i++) {
			while (copy_sign.indexOf(order[i]) !== -1) {
				let idx = copy_sign.indexOf(order[i]);

				if (order[i] === "+") {
					copy_num[idx] += copy_num[idx + 1];
				} else if (order[i] === "-") {
					copy_num[idx] -= copy_num[idx + 1];
				} else if (order[i] === "*") {
					copy_num[idx] *= copy_num[idx + 1];
				}

				// 계산 후 인덱스로 배열 삭제
				copy_sign.splice(idx, 1);
				copy_num.splice(idx + 1, 1);
			}
		}

		answer = Math.max(answer, Math.abs(copy_num[0]));
	});

	return answer;
}
